src/errors/register.lua

```diff
@@ -7,20 +9,21 @@
 --- Registers a new error class
 ---@param name string
 ---@param constructor fun(...: any): ErrorBody
+---@param metatable? table<string, any> Optional custom metatable
```

Comment:
Do we need to constrain the key to `string`? I think just having this as `table` would be fine.